prost-build/src/context.rs

```diff
@@ -0,0 +1,285 @@
+use std::borrow::Cow;
+
+use prost_types::{
+    field_descriptor_proto::{Label, Type},
+    FieldDescriptorProto,
+};
+
+use crate::extern_paths::ExternPaths;
+use crate::message_graph::MessageGraph;
+use crate::{BytesType, Config, MapType, ServiceGenerator};
+
+/// The context providing all the global information needed to generate code.
+/// It also provides a more disciplined access to Config
+/// and its mutable instance of ServiceGenerator.
+pub struct Context<'a> {
+    config: &'a mut Config,
```

Comment:
In the PR description, you mention that a mutable Config is awkward. But it is still `mut` can you further explain what has been improved?